homeworks/homework2/bank/src/main/java/com/rtejada/bank/service/TransferService.java

```diff
@@ -0,0 +1,79 @@
+package com.rtejada.bank.service;
+
+import com.rtejada.bank.exception.InvalidTransactionException;
+import com.rtejada.bank.model.Account;
+import com.rtejada.bank.model.AccountType;
+import com.rtejada.bank.model.Transfer;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.rtejada.bank.model.AccountType.CREDIT;
+import static com.rtejada.bank.model.AccountType.SAVING;
+
+@Service
+public class TransferService {
+
+	private final Map<AccountType, AccountService> accountServiceMap;
+	private static final BigDecimal INITIAL_TRANSFER_FEE = BigDecimal.valueOf(1.02);
+	private static final BigDecimal TRANSFER_FEE = BigDecimal.valueOf(1.05);
+	private static final Map<Long, BigDecimal> TRANSFER_RULES = buildTransferRules();
+
+	public TransferService(CreditAccountService creditAccountService, SaveAccountService saveAccountService) {
+		this.accountServiceMap = new HashMap<>();
+		accountServiceMap.put(CREDIT, creditAccountService);
+		accountServiceMap.put(SAVING, saveAccountService);
+	}
+
+	public BigDecimal transfer(Transfer transfer) {
+		BigDecimal amountWithFee;
+
+		if (SAVING.equals(transfer.getSourceType()) || SAVING.equals(transfer.getTargetType())) {
+
+			final Account sourceAccount = accountServiceMap.get(transfer.getSourceType()).getAccount(transfer.getSourceId());
+
+			String sourceCpf = accountServiceMap.get(transfer.getSourceType()).getAccount(transfer.getSourceId()).getOwner().getCpf();
```

Comment:
Este trecho de código está repetido: ` accountServiceMap.get(transfer.getSourceType()).getAccount(transfer.getSourceId())`.
```suggestion
			String sourceCpf = sourceAccount.getOwner().getCpf();
```

Reply:
feito